python/cudf/bindings/csv.pyx

```diff
@@ -0,0 +1,423 @@
+        decompression). If using ‘zip’, the ZIP file must contain only one
+        data file to be read in, otherwise the first non-zero-sized file will
+        be used. Set to None for no decompression.
+    decimal : char, default '.'
+        Character used as a decimal point.
+    thousands : char, default None
+        Character used as a thousands delimiter.
+    true_values : list, default None
+        Values to consider as boolean True
+    false_values : list, default None
+        Values to consider as boolean False
+    nrows : int, default None
+        If specified, maximum number of rows to read
+    byte_range : list or tuple, default None
+        Byte range within the input file to be read. The first number is the
+        offset in bytes, the second number is the range size in bytes. Set the
+        size to zero to read all data after the offset location. Reads the row
+        that starts before or at the end of the range, even if it ends after
+        the end of the range.
+    skip_blank_lines : bool, default True
+        If True, discard and do not parse empty lines
+        If False, interpret empty lines as NaN values
+    comment : char, default None
+        Character used as a comments indicator. If found at the beginning of a
+        line, the line will be ignored altogether.
+    na_values : list, default None
+        Values to consider as invalid
+    keep_default_na : bool, default True
+        Whether or not to include the default NA values when parsing the data.
+    na_filter : bool, default True
+        Detect missing values (empty strings and the values in na_values).
+        Passing False can improve performance.
+    prefix : str, default None
+        Prefix to add to column numbers when parsing without a header row
+    index_col : int or string, default None
+        Column to use as the row labels
+
+    Returns
+    -------
+    GPU ``DataFrame`` object.
+
+    Examples
+    --------
+
+    Create a test csv file
+
+    >>> import cudf
+    >>> filename = 'foo.csv'
+    >>> lines = [
+    ...   "num1,datetime,text",
+    ...   "123,2018-11-13T12:00:00,abc",
+    ...   "456,2018-11-14T12:35:01,def",
+    ...   "789,2018-11-15T18:02:59,ghi"
+    ... ]
+    >>> with open(filename, 'w') as fp:
+    ...     fp.write('\\n'.join(lines)+'\\n')
+
+    Read the file with ``cudf.read_csv``
+
+    >>> cudf.read_csv(filename)
+      num1                datetime text
+    0  123 2018-11-13T12:00:00.000 5451
+    1  456 2018-11-14T12:35:01.000 5784
+    2  789 2018-11-15T18:02:59.000 6117
+    """
+
+    if delim_whitespace:
+        if delimiter is not None:
+            raise ValueError("cannot set both delimiter and delim_whitespace")
+        if sep != ',':
+            raise ValueError("cannot set both sep and delim_whitespace")
+
+    # Alias sep -> delimiter.
+    if delimiter is None:
+        delimiter = sep
+
+    if dtype is not None:
```

Comment:
```suggestion
    dtype_dict = False
    if dtype is not None:
```
I think we need to initialize `dtype_dict`, otherwise we get this warning when compiling Cython because of later usage (from what I can tell from the generated .cpp code):
```bash
cudf/bindings/csv.cpp:3142:9: warning: ‘__pyx_v_dtype_dict’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         if (__pyx_t_5) {
         ^~
cudf/bindings/csv.cpp:1911:7: note: ‘__pyx_v_dtype_dict’ was declared here
   int __pyx_v_dtype_dict;
       ^~~~~~~~~~~~~~~~~~
```